pkg/remote/codec/thrift/thrift.go

```diff
@@ -108,6 +108,11 @@ func (c thriftCodec) Marshal(ctx context.Context, message remote.Message, out re
 		}
 	}
 
+	if hyperMarshalFallback(data) {
+		// fallback to frugal
+		return c.hyperMarshal(out, methodName, msgType, seqID, data)
+	}
+
 	// fallback to old thrift way (slow)
 	return encodeBasicThrift(out, ctx, methodName, msgType, seqID, data)
```

Comment:
fallback should be after `encodeBasicThrift` returning an error indicating it's not supported (for codegen with slim template)

Reply:
fixed.